packages/core/src/main/util/LowPrecisionTaskScheduler/LowPrecisionTaskScheduler.test.node.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {LowPrecisionTaskScheduler} from './LowPrecisionTaskScheduler';
+
+describe('LowPrecisionTaskScheduler', () => {
+  beforeEach(() => {
+    jasmine.clock().install();
+    jasmine.clock().mockDate(new Date(0));
+  });
+  afterEach(() => {
+    jasmine.clock().uninstall();
+  });
+
+  it("task won't run again after it was executed previously", async () => {
+    const mockedTask = jasmine.createSpy('mockedTask', () => Promise.resolve('hello task'));
+
+    LowPrecisionTaskScheduler.addTask({
+      key: 'test-key',
+      firingDate: 0,
+      intervalDelay: 2000,
+      task: mockedTask,
+    });
+
+    jasmine.clock().tick(2001);
+    await Promise.resolve();
+
+    jasmine.clock().tick(2001);
+    await Promise.resolve();
+
+    expect(mockedTask).toHaveBeenCalledTimes(1);
+  });
+
+  it('adds single task to schedule and runs it after given delay', async () => {
+    const mockedTask = jasmine.createSpy('mockedTask', () => Promise.resolve('hello task'));
+
+    LowPrecisionTaskScheduler.addTask({
+      key: 'test-key',
+      firingDate: 0,
+      intervalDelay: 1000,
+      task: mockedTask,
+    });
+
+    jasmine.clock().tick(1001);
+    await Promise.resolve();
```

Comment:
Can't find other way to make scheduled async task execute before `expect` check... Any ideas? @atomrc 😄